src/sqliteDB/sql.cpp

```diff
@@ -222,22 +209,24 @@ int Database::executeCommand(std::string command, std::string errMsg,
         std::cerr << "Error in SQL execute: Only type 1 and 0 should exist!"
           << std::endl;
         std::cerr << errMsg << std::endl;
-        return 1;
+        return -1;
     }
 
-    if (exit != SQLITE_OK) {
-        // Looks like messageError is not guaranteed to be set even if there is
-        // an error.
-        if (messageError) {
-          std::cerr << errMsg << ": " << messageError << std::endl;
-          sqlite3_free(messageError);
-        } else {
-          std::cerr << errMsg << std::endl;
-        }
-        return 1;
+    if (exit == 19) {
+        std::cerr << "ERROR_CODE: 19, Constraint/Duplicate Error!" <<
+          std::endl;
+        sqlite3_free(messageError);
+        return 0;
+    } else if (exit != SQLITE_OK) {
```

Comment:
We cannot depend on messageError being set by sqlite3_exec(), even if there is an error. To avoid potential segfaults, only print and free messageError if it is not null as done on trunk

Reply:
Also it looks like some static analyzer issues popped up based on the merge.
Outside of these two issues code looks good

Reply:
Re-implemented fixed code inside the second else - if conditional